python/whylogs/api/logger/experimental/logger/actor/process_rolling_logger_messages.py

```diff
@@ -0,0 +1,283 @@
+TypedDict or wrappers around serialized byte versions of those dicts. Everything in here is a
+TypedDict because we use orjson to deserialize them for performance reasons and that library prefers
+to output everything as dictionaries.
+
+The dataclass containers for those messages have helper methods to extract/deserialize the dicts. It's
+important to not raise exceptions in those data classes because the actor does a lot of large batch processing
+and exceptions would result in losing the entire batch. Instead, they return None and log errors if there is
+some issue deserializing or validating.
+"""
+import base64
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple, TypedDict, Union, cast
+
+from whylogs.api.logger.experimental.logger.actor.thread_rolling_logger import (
+    LoggerStatus,
+)
+
+try:
+    import orjson
+except ImportError:
+    raise ImportError("Install whylogs with extra [proc] for process based logging: pip install whylogs[proc]")
+import logging
+import uuid
+
+import numpy as np
+import pandas as pd
+
+from whylogs.api.logger.experimental.logger.actor.time_util import (
+    TimeGranularity,
+    truncate_time_ms,
+)
+
+DataTypes = Union[str, int, float, bool, List[float], List[int], List[str]]
+
+_logger = logging.getLogger("profile_actor_messages")
+
+
+class DataDict(TypedDict):
+    columns: List[str]
+    data: List[List[DataTypes]]
+
+
+def data_dict_from_pandas(df: pd.DataFrame) -> DataDict:
+    multiple = df.to_dict(orient="split")
+    del multiple["index"]  # get rid of this to min/max payload size
+    return {
+        "columns": multiple["columns"],
+        "data": multiple["data"],
+    }
+
+
+class LogRequestDict(TypedDict):
+    datasetId: str
+    timestamp: Optional[int]
+    multiple: DataDict
+
+
+class LogEmbeddingRequestDict(TypedDict):
+    datasetId: str
+    timestamp: Optional[int]
+    embeddings: Dict[str, List[DataTypes]]
+
+
+class PubSubMessage(TypedDict):
+    attributes: Dict[str, str]
+    data: str
+    message_id: str
+    publish_time: str
+
+
+class PubSubDict(TypedDict):
+    subscription: str
+    message: PubSubMessage
+    log_request: LogRequestDict
+
+
+class DebugMessage:
+    pass
+
+
+class FlushMessage:
+    pass
+
+
+@dataclass
+class ProcessLoggerStatus:
+    dataset_id: str
+    status: LoggerStatus
+
+
+@dataclass
+class ProcessLoggerStatusMessage:
+    id: str = "None"
+
+    def __post_init__(self) -> None:
+        if self.id == "None":
+            self.id = str(uuid.uuid4())
+
+
+@dataclass
+class LogMessage:
+    request_time: int
+    log: LogRequestDict
+    id: str = "None"
+
+    def __post_init__(self) -> None:
+        if self.id == "None":
+            self.id = str(uuid.uuid4())
+
+
+@dataclass
+class RawLogMessage:
+    request: bytes
+    """
+    Bytes that represent json stringified LogRequestDict
+    """
+    request_time: int
+    id: str = "None"
```

Comment:
why not `id = field(default=None)` + `__post_init__`? little weird to have a string "None" in all of them

Reply:
default wouldn't have worked because it makes the type optional, which I can't have. But there's also a default_factory which was exactly what I needed.